src/test/java/io/burt/athena/support/QueryExecutionHelper.java

```diff
@@ -143,28 +148,24 @@ public void queueGetQueryExecutionException(Exception e) {
         if (delay.isZero()) {
             return future;
         } else {
-            CompletableFuture<T> newFuture = new CompletableFuture<>();
-            ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
-            scheduler.schedule(
-                    () -> {
-                        try {
-                            newFuture.complete(future.get());
-                        } catch (ExecutionException e) {
-                            newFuture.completeExceptionally(e.getCause());
-                        } catch (Exception e) {
-                            newFuture.completeExceptionally(e);
-                        } finally {
-                            clock.tick(delay);
-                        }
-                    },
-                    delay.toMillis(),
-                    TimeUnit.MILLISECONDS
-            );
-            scheduler.shutdown();
-            return newFuture;
+            TestDelayedCompletableFuture<T> testFuture = new TestDelayedCompletableFuture<>(future, delay, clock);
+            CompletableFuture<T> restrictedFuture = mockCompletableFuture(invocation -> {
+                throw new UnsupportedOperationException(invocation.getMethod().toString());
+            });
+            try {
+                doAnswer(AdditionalAnswers.delegatesTo(testFuture)).when(restrictedFuture).get(anyLong(), any());
+            } catch (ExecutionException | InterruptedException | TimeoutException e) {
+                e.printStackTrace();
```

Comment:
This behaviour is different from before. Would it be possible to propagate the error instead of dropping it on the floor? I don't know if any of these can actually happen, or are just part of the signature (but `throw new RuntimeException(e)` somehow feels better than print-and-continue).

Reply:
Actually, we are not. This is just an unfortunate consequence of checked exceptions and the fact that Mockito provides an object that simulates the original class when specifying the behavior. Expressed differently, none of these exceptions will ever be thrown here (assuming Mockito only throws non-checked exceptions when defining behavior actually fails). The only reason there is code here at all, was that it felt dirty leaving the block empty, but I suppose your suggestion is about as much code, and would provide a better experience if Mockito would for some mysterious reason thrown one of these errors.